### PR TITLE
2.0

### PR DESCRIPTION
**Pull request by fabpot**
The main ideas of 2.0:

 * Possibility to have one main checkout of this repository and be able to check many projects without using Composer
 * Possibility to check Symfony requirements without a project: `php bin/requirements-checker.php` or for a specific project `php bin/requirements-checker.php project_dir` (or when used as a Composer dep)

### CODE REVIEW

File: bin/requirements-checker.php [hunk 1]
```diff
@@ -106,13 +120,11 @@
...
+echo '  The command console can use a different php.ini file'.PHP_EOL;
 echo_style('title', '~~~~');
-echo '  than the one used with your web server. To be on the'.PHP_EOL;
-echo '      safe side, please check the requirements from your web'.PHP_EOL;
-echo '      server using the ';
-echo_style('yellow', 'public/check.php');
-echo ' script.'.PHP_EOL;
+echo '  than the one used by your web server.'.PHP_EOL;
+echo '      Please check that both the console and the web server'.PHP_EOL;
+echo '      are using the same PHP version and configuration.'.PHP_EOL;
```
Comment (fabpot) [inline bin/requirements-checker.php:127]: I've removed the `public/check.php` file as I think the better recommendation is to be sure that both the console and the web server are using the same PHP version and configuration (at least for everything that we check).

File: bin/requirements-checker.php [hunk 2]
```diff
@@ -17,15 +17,29 @@
 } elseif (file_exists($autoloader = __DIR__.'/../vendor/autoload.php')) {
     require_once $autoloader;
 } else {
-    throw new \RuntimeException('Unable to find the Composer autoloader.');
+    require_once dirname(__DIR__).'/src/Requirement.php';
+    require_once dirname(__DIR__).'/src/RequirementCollection.php';
+    require_once dirname(__DIR__).'/src/PhpConfigRequirement.php';
+    require_once dirname(__DIR__).'/src/SymfonyRequirements.php';
```
Comment (fabpot) [inline bin/requirements-checker.php:23]: This allows to use it without having to require it in a project.

File: bin/requirements-checker.php [hunk 3]
```diff
@@ -17,15 +17,29 @@
...
+    } else {
+        $args[] = $arg;
+    }
+}
 
 $symfonyVersion = class_exists('\Symfony\Component\HttpKernel\Kernel') ? \Symfony\Component\HttpKernel\Kernel::VERSION : null;
 
-$symfonyRequirements = new SymfonyRequirements(dirname(dirname(realpath($autoloader))), $symfonyVersion);
+// specific directory to check?
+$dir = isset($args[1]) ? $args[1] : dirname(dirname(realpath($autoloader)));
```
Comment (fabpot) [inline bin/requirements-checker.php:40]: This allows to have only one checkout of the repository and run it on different directories.

**Comment by jorrit** [comment]
@fabpot Can you check https://github.com/symfony/recipes/pull/821 ? The current 1.0 recipe doesn't work anymore.

**Comment by TELLO0815** [comment]
Same question, I don't find any information how to check with v2???

**Pull request merged**